Feature/lit 2726 Lit Action Plus Documentation (#274)
LIT-Protocol/docs: docs/sdk/serverless-signing/run-once.md

import FeedbackComponent from "@site/src/pages/feedback.md";

# Run on a Single Node Within an Action

## Overview

> Witin a Lit action, you may want only a single node to perform a specific piece of your lit action logic, and let the result be broadcast to all other nodes which are executing the same Lit action in parallel. `runOnce` takes a function as a parameter, and all the nodes use a deterministic algorithm to choose which node will run the function that is passed to runOnce. This single node runs the function, and then broadcasts the result to all the other nodes.

Reviewer: Copy suggestion:

Current:
"Witin a Lit action, you may want only a single node to perform a specific piece of your lit action logic, and let the result be broadcast to all other nodes which are executing the same Lit action in parallel. `runOnce` takes a function as a parameter, and all the nodes use a deterministic algorithm to choose which node will run the function that is passed to runOnce. This single node runs the function, and then broadcasts the result to all the other nodes."

Suggestion:
"Typically, when a Lit Action is called it is executed across every Lit node in parallel. With `runOnce`, you have the ability to perform specified operations on a _single_ node, versus all of them at once. 

The `runOnce` function takes another function(s) as a parameter and a deterministic algorithm is used to select the Lit node that it will be executed on. This node will run the function and broadcast the result to all of the other Lit nodes."

import FeedbackComponent from "@site/src/pages/feedback.md";

# Run on a Single Node Within an Action

## Overview

> Typically, when a Lit Action is called it is executed across every Lit node in parallel. With runOnce, you have the ability to perform specified operations on a single node, versus all of them at once.

Reviewer 2: ```suggestion
Typically, when a Lit Action is called it is executed across every Lit node in parallel. With `runOnce`, you have the ability to perform specified operations on a single node, versus all of them at once.
```

> The runOnce function takes another function(s) as a parameter and a deterministic algorithm is used to select the Lit node that it will be executed on. This node will run the function and broadcast the result to all of the other Lit nodes.

Reviewer 2: ```suggestion
The `runOnce` function takes another function(s) as a parameter and a deterministic algorithm is used to select the Lit node that it will be executed on. This node will run the function and broadcast the result to all of the other Lit nodes.
```

    let hash = utils.keccak256(ethers.utils.toUtf8Bytes(serializedTx));
    // encode the message into an uint8array for signing
    const toSign = await new TextEncoder().encode(hash);
    const signature = await Lit.Actions.signAndCombineEcdsa({
        toSign,
        publicKey,
        sigName,
    });

    // the code in the function given to runOnce below will only be run by one node
    let res = await Lit.Actions.runOnce({ waitForResponse: true, name: "txnSender" }, async () => {
        // get the node operator's rpc url for the 'ethereum' chain
        const rpcUrl = await Lit.Actions.getRpcUrl({ chain: "ethereum" });
        const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
        const tx = await provider.sendTransaction(signature);
        return tx; // return the tx to be broadcast to all other nodes
    });

    // set the response from the action as the result of runOnce operation
    // will be sent by all nodes, even though only a single node did the computation
    Lit.Actions.setResponse(res);
})()
`
const client = new LitNodeClient({
    litNetwork: 'cayenne',
});

await client.connect();

> const res = client.executeJs({

Reviewer 2: ```suggestion
const res = await client.executeJs({
```

    code,
    sessionSigs: {} // your session
    jsParams: {
        publicKey: "<your pkp public key>",
    }
});

console.log("transactions in latest block from all nodes: ", res);
```

> In the above `runOnce` example you'll notice we provide two properties in the `param object`

Reviewer 2: ```suggestion
In the above `runOnce` example, within the Lit Action code, you'll notice we specify two properties in the object passed to `Lit.Actions.runOnce`:
```
